### ISSUE TITLE
Wait until the AVD instance is running

### ISSUE THREAD

**Issue by progxaker** [FreeRDP/FreeRDP#9214]
**Is your feature request related to a problem? Please describe.**
I use AVD with auto-shutdown plugin and when the host is down,
I get the following error:

<details>
<summary>Error log</summary>

```
Browse to: https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=a85cf173-4192-42f8-81fa-777a763e6e2c&response_type=code&scope=https%3A%2F%2Fwww.wvd.microsoft.com%2F.default&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient  
Paste redirect URL here: 
https://login.microsoftonline.com/common/oauth2/nativeclient?code=<code>&session_state=<state>
[09:32:27:253] [35758:00008bb0] [DEBUG][com.freerdp.utils.http] - [freerdp_http_request]: headers:
POST /common/oauth2/v2.0/token HTTP/1.1
Host: login.microsoftonline.com
Content-Type: application/x-www-form-urlencoded
Content-Length: 1039


[09:32:28:453] [35758:00008bb0] [DEBUG][com.freerdp.utils.http] - [freerdp_http_request]: body:
grant_type=authorization_code&code=<code>&client_id=a85cf173-4192-42f8-81fa-777a763e6e2c&scope=https%3A%2F%2Fwww.wvd.microsoft.com%2F.default&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient
[09:32:28:30] [35758:00008bb0] [DEBUG][com.freerdp.utils.ringbuffer] - [ringbuffer_write]: ringbuffer_write(0x60600001e330): sz: 4841
[09:32:28:30] [35758:00008bb0] [DEBUG][com.freerdp.utils.ringbuffer] - [ringbuffer_peek]: ringbuffer_peek(0x60600001e330): sz: 4841
[09:32:28:30] [35758:00008bb0] [DEBUG][com.freerdp.utils.ringbuffer] - [ringbuffer_commit_read_bytes]: ringbuffer_commit_read_bytes(0x60600001e330): sz: 4841
[09:32:28:30] [35758:00008bb0] [DEBUG][com.freerdp.utils.ringbuffer] - [ringbuffer_write]: ringbuffer_write(0x60600001e330): sz: 238
[09:32:28:30] [35758:00008bb0] [DEBUG][com.freerdp.utils.ringbuffer] - [ringbuffer_peek]: ringbuffer_peek(0x60600001e330): sz: 238
[09:32:28:30] [35758:00008bb0] [DEBUG][com.freerdp.utils.ringbuffer] - [ringbuffer_commit_read_bytes]: ringbuffer_commit_read_bytes(0x60600001e330): sz: 238
[09:32:29:872] [35758:00008bb0] [DEBUG][com.freerdp.core.gateway.http] - [http_chuncked_read]: chunked encoding end of stream received
[09:32:29:872] [35758:00008bb0] [ERROR][com.freerdp.core.gateway.arm] - [arm_resolve_endpoint]: Unexpected HTTP status: HTTP_STATUS_BAD_REQUEST [400]
[09:32:29:872] [35758:00008bb0] [DEBUG][com.freerdp.utils.ringbuffer] - [ringbuffer_write]: ringbuffer_write(0x60600001e330): sz: 24
[09:32:29:872] [35758:00008bb0] [DEBUG][com.freerdp.utils.ringbuffer] - [ringbuffer_peek]: ringbuffer_peek(0x60600001e330): sz: 24
[09:32:29:872] [35758:00008bb0] [DEBUG][com.freerdp.utils.ringbuffer] - [ringbuffer_commit_read_bytes]: ringbuffer_commit_read_bytes(0x60600001e330): sz: 24
[09:32:29:873] [35758:00008bb0] [DEBUG][com.freerdp.utils.ringbuffer] - [ringbuffer_destroy]: ringbuffer_destroy(0x60600001e330)
[09:32:29:873] [35758:00008bb0] [ERROR][com.freerdp.core.nego] - [nego_connect]: Protocol Security Negotiation Failure
[09:32:29:873] [35758:00008bb0] [ERROR][com.freerdp.core] - [rdp_client_connect]: ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x0002000C]
[09:32:29:873] [35758:00008bb0] [ERROR][com.freerdp.core.connection] - [rdp_client_connect]: Error: protocol security negotiation or connection failure
[09:32:29:875] [35758:00008bb0] [DEBUG][com.freerdp.core.rdp] - [rdp_finalize_reset_flags][0x61f000000080]: [CONNECTION_STATE_NEGO] reset finalize_sc_pdus
[09:32:29:875] [35758:00008bb0] [DEBUG][com.freerdp.core.connection] - [rdp_client_transition_to_state]: CONNECTION_STATE_NEGO --> CONNECTION_STATE_INITIAL
```

</details>

<details>
<summary>MITMProxy</summary>

```
POST /api/arm/v2/connections/ HTTP/1.1
Cache-Control: no-cache
Pragma: no-cache
Accept: application/json
User-Agent: FreeRDP/3.0
Host: afdfp-rdgateway-r1.wvd.microsoft.com
Connection: Keep-Alive
Content-Length: 216
Authorization: Bearer <bearer-token>
Content-Type: application/json
X-MS-User-Agent: FreeRDP/3.0

{"application":"||<application>","loadBalanceInfo":"mth://localhost/<id>/<id>","LogonToken":null,"gatewayLoadBalancerToken":null}

HTTP/1.1 400 Bad Request
Transfer-Encoding: chunked
Content-Type: application/json; charset=utf-8
Set-Cookie: TiPMix=79.9614359065711; path=/; HttpOnly; Domain=rdgateway-c101-gwc-r1.wvd.microsoft.com; Max-Age=3600; Secure; SameSite=None
Set-Cookie: x-ms-routing-name=self; path=/; HttpOnly; Domain=rdgateway-c101-gwc-r1.wvd.microsoft.com; Max-Age=3600; Secure; SameSite=None
Set-Cookie: ARRAffinity=<arr-affinity>;Path=/;HttpOnly;Secure;Domain=rdgateway-c101-gwc-r1.wvd.microsoft.com
Set-Cookie: ARRAffinitySameSite=<arr-affinity-same-site>;Path=/;HttpOnly;SameSite=None;Secure;Domain=rdgateway-c101-gwc-r1.wvd.microsoft.com
api-supported-versions: 2.0
X-Rate-Limit-Limit: <limit-limit>
X-Rate-Limit-Remaining: <limit-remaining>
X-Rate-Limit-Reset: <limit-reset>
x-ms-wvd-service-region: <wvd-service-region>
x-ms-opsarmpath64: <opsarmpath64>
x-ms-lamport-ts: <lamport-ts>
x-ms-correlation-id: <correlation-id>
x-ms-request-id: <request-id>
X-Powered-By: ASP.NET
X-Content-Type-Options: nosniff
X-Cache: CONFIG_NOCACHE
X-MSEdge-Ref: Ref A: <ref-a> Ref B: <ref-b> Ref C: <ref-c>
Date: <date>

{
  "Code": "E_PROXY_ORCHESTRATION_LB_SESSIONHOST_DEALLOCATED",
  "Message": "Session Host with ID = '<host-id>' has been deallocated we are starting the session host for you. Please retry connecting after 5 minutes.",
  "Target": null,
  "Details": null,
  "InnerError": null
}
```

</details>

**Describe the solution you'd like**
The Web client checks the connection every minute with the same request.

**Comment by akallabeth** [comment]
out of scope, the target must be running to connect

**Comment by progxaker** [comment]
Do I understand correctly that FreeRDP will also not support reconnect/restore session?

**Comment by akallabeth** [comment]
if there is a RDP channel for that then we can add support, but as far as I know this is out of scope for RDP but a different mechanism.

**Comment by progxaker** [comment]
1) The web client requests `/api/arm/v2/connections` and get the error.
1.1) The client shows like `Starting VM. Please wait up to 5 minutes`.
2) In parallel, the server (not sure if it's Azure or RDP Gateway) starts AVD.
3) After 1 minute, the client request `/api/arm/v2/connections`, get the error.
4) After 1 minute, the client request `/api/arm/v2/connections`, get the error.
5) After 1 minute, the client request `/api/arm/v2/connections`, get 200 and start connecting (FreeRDP can also connect).

**Comment by akallabeth** [comment]
so, basically try reconnecting until it is started up?

**Comment by progxaker** [comment]
> so, basically try reconnecting until it is started up?

Yes, until the timeout expires (assumed 5 minutes). I can't get an error on the timeout so far, but I assume it looks like "Please contact your administrator."

**Comment by progxaker** [comment]
Hello @akallabeth. I made a code change just as Proof of Concept. I might want to create a finished PR, but I current don't have the time.

I've been testing this for two weeks and haven't experienced any problems. There are many things that should have been applied (proper logging, function exits, moving the loop to a separate function), but this is PoC :)

```diff
diff --git a/libfreerdp/core/gateway/arm.c b/libfreerdp/core/gateway/arm.c
index 1d2badbc2..088c6aae7 100644
--- a/libfreerdp/core/gateway/arm.c
+++ b/libfreerdp/core/gateway/arm.c
@@ -56,6 +56,9 @@
 #include <cjson/cJSON.h>
 #endif
 
+#include <time.h>
+#include <string.h>
+
 struct rdp_arm
 {
 	rdpContext* context;
@@ -65,6 +68,8 @@ struct rdp_arm
 
 typedef struct rdp_arm rdpArm;
 
+short int gateway_retry = 0;
+
 #define TAG FREERDP_TAG("core.gateway.arm")
 
 #ifdef WITH_AAD
@@ -829,6 +834,69 @@ BOOL arm_resolve_endpoint(rdpContext* context, DWORD timeout)
 		if (!res)
 			goto arm_error;
 	}
+	else if (StatusCode == HTTP_STATUS_BAD_REQUEST)
+	{
+		const size_t response_length = http_response_get_body_length(response);
+		char* msg = calloc(response_length + 1, sizeof(char));
+		if (!msg)
+			goto arm_error;
+
+		memcpy(msg, http_response_get_body(response), response_length);
+		WLog_DBG(TAG, "Got HTTP Response data: %s", msg);
+
+		cJSON *json = cJSON_ParseWithLength((const char*)msg, response_length);
+		if (json == NULL)
+		{
+			const char *error_ptr = cJSON_GetErrorPtr();
+			if (error_ptr != NULL)
+			{
+				WLog_ERR(TAG, "NullPoException: %s", error_ptr);
+				free(msg);
+				goto arm_error;
+			}
+		}
+
+                const cJSON *gateway_code_str = NULL;
+                gateway_code_str = cJSON_GetObjectItemCaseSensitive(json, "Code");
+		if (!cJSON_IsString(gateway_code_str) || (gateway_code_str->valuestring == NULL))
+		{
+			WLog_ERR(TAG, "Response has no \"Code\" property");
+			http_response_log_error_status(WLog_Get(TAG), WLOG_ERROR, response);
+			if (json) {
+				cJSON_Delete(json);
+			}
+			free(msg);
+			goto arm_error;
+		}
+
+		if (strcmp(gateway_code_str->valuestring, "E_PROXY_ORCHESTRATION_LB_SESSIONHOST_DEALLOCATED") == 0) {
+			int max_retries = 10;
+			int code_execution_interval = 60;
+
+			if (gateway_retry >= max_retries) {
+				WLog_ERR(TAG, "Timeout reached. Exiting loop.");
+				if (json) {
+					cJSON_Delete(json);
+				}
+				free(msg);
+				goto arm_error;
+			}
+			WLog_INFO(TAG, "Please wait... VM is starting running.");
+			if (json) {
+				cJSON_Delete(json);
+			}
+			free(msg);
+			sleep(code_execution_interval);
+			arm_resolve_endpoint(context, timeout);
+		}
+		else
+		{
+			cJSON_Delete(json);
+			free(msg);
+			http_response_log_error_status(WLog_Get(TAG), WLOG_ERROR, response);
+			goto arm_error;
+		}
+	}
 	else
 	{
 		http_response_log_error_status(WLog_Get(TAG), WLOG_ERROR, response);
```

**Comment by akallabeth** [comment]
@progxaker better create a (draft) pull request, easier to fix stuff then.

**Comment by progxaker** [comment]
Thank you @akallabeth and @hardening! Honestly, you did all the work :')

**Comment by akallabeth** [comment]
@progxaker well, hope it works?

**Comment by progxaker** [comment]
I'll try tomorrow morning and let you know.

**Comment by progxaker** [comment]
I built the FreeRDP package from the `master` branch and updated the rdpw file from Web-client, but got the following error after providing the redirect URL.
`[17:53:34:608] [19587:00004c84] [ERROR][com.freerdp.core.connection] - [rdp_client_connect]: error retrieving ARM configuration`

**Comment by akallabeth** [comment]
@progxaker can you check where in `arm_resolve_endpoint` the error occurs by stepping through a debugger?

**Comment by progxaker** [comment]
I removed the following part and suspect that `len` must be one more
```diff
static BOOL arm_handle_request_ok(rdpArm* arm, const HttpResponse* response)
{
        const size_t len = http_response_get_body_length(response);
        const char* msg = http_response_get_body(response);
-        if (strnlen(msg, len) >= len)
-              return FALSE;

        WLog_DBG(TAG, "Got HTTP Response data: %s", msg);
        return arm_fill_gateway_parameters(arm, msg, len);
}
```
because  it was previously:
```
		const size_t len = http_response_get_body_length(response);
		char* msg = calloc(len + 1, sizeof(char));
		if (!msg)
			goto arm_error;

		memcpy(msg, http_response_get_body(response), len);
```

**Comment by akallabeth** [comment]
@progxaker fixed with #9390

**Comment by akallabeth** [comment]
@progxaker changed the logic to respect `/auto-reconnect` setting for retries.

**Comment by progxaker** [comment]
Oh, I saw about AutoReconnectionEnabled, but didn't know it was about the parameter.
But anyway, I didn't set it, so FreeRDP shouldn't try to reconnect, but it did and couldn't when AVD is up)

**Comment by progxaker** [comment]
The reconnects work as expected with `/auto-reconnect`.

```
[10:43:41:986] [74218:000121eb] [INFO][com.freerdp.client.common] - [client_common_retry_dialog]: [arm-transport] Starting your VM. It may take up to 5 minutes
[10:43:41:986] [74218:000121eb] [INFO][com.freerdp.client.common] - [client_common_retry_dialog]: [arm-transport] retry 0/20, delaying 15000ms before next attempt
[10:43:57:660] [74218:000121eb] [INFO][com.freerdp.client.common] - [client_common_retry_dialog]: [arm-transport] retry 1/20, delaying 15000ms before next attempt                        
[10:44:12:192] [74218:000121eb] [INFO][com.freerdp.client.common] - [client_common_retry_dialog]: [arm-transport] retry 2/20, delaying 15000ms before next attempt                           
[10:44:28:764] [74218:000121eb] [INFO][com.freerdp.client.common] - [client_common_retry_dialog]: [arm-transport] retry 3/20, delaying 15000ms before next attempt
[10:44:44:874] [74218:000121eb] [INFO][com.freerdp.client.common] - [client_common_retry_dialog]: [arm-transport] retry 4/20, delaying 15000ms before next attempt
```

**Comment by akallabeth** [comment]
@progxaker thx for confirmation. looks good from our side too (configurable, but no new/duplicated options - a real horror maintaining such if they are not properly defined)

**Comment by progxaker** [comment]
> But anyway, I didn't set it, so FreeRDP shouldn't try to reconnect, but it did and couldn't when AVD is up)

Yes, FreeRDP try to reach the host *every* second w/o `/auto-reconnect`.

**Comment by akallabeth** [comment]
> > But no new/duplicated options - a real horror maintaining such if they are not properly defined
> 
> Heh, I know that pain)
> 
> > But anyway, I didn't set it, so FreeRDP shouldn't try to reconnect, but it did and couldn't when AVD is up)
> 
> Yes, FreeRDP try to reach a disabled host _every_ second w/o `/auto-reconnect`.

@progxaker sorry, but what? you confused me with the last comment

**Comment by progxaker** [comment]
> sorry, but what? you confused me with the last comment

Hehe 😅. FreeRDP works fine with the `/auto-reconnect` parameter, but if I remove it, FreeRDP tries to connect to a disabled AVD anyway each second.

<details>
<summary>Debug logs</summary>

```
# sdl-freerdp ./SessionDesktop.rdpw /gateway:type:arm /f /size:1366x768 /u:<email> /p:'<password>' /mic:sys:pulse,format:1 /network:auto /log-level:DEBUG
...
[10:10:07:593] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_connect_begin]: resetting error state
[10:10:07:810] [89418:00015d4b] [DEBUG][com.freerdp.client.common.cmdline] - [freerdp_client_load_static_channel_addin]: loading channelEx rdpdr
[10:10:07:810] [89418:00015d4b] [DEBUG][com.freerdp.client.common.cmdline] - [freerdp_client_load_static_channel_addin]: loading channelEx rdpsnd
[10:10:07:810] [89418:00015d4b] [DEBUG][com.freerdp.channels.drdynvc.client] - [drdynvc_VirtualChannelEntryEx]: VirtualChannelEntryEx
[10:10:07:810] [89418:00015d4b] [DEBUG][com.freerdp.client.common.cmdline] - [freerdp_client_load_static_channel_addin]: loading channelEx drdynvc
[10:10:07:825] [89418:00015d4b] [DEBUG][com.freerdp.primitives] - [primitives_autodetect_best]: primitives benchmark result:
[10:10:08:998] [89418:00015d4b] [DEBUG][com.freerdp.primitives] - [primitives_autodetect_best]:  * generic= 14
[10:10:08:158] [89418:00015d4b] [DEBUG][com.freerdp.primitives] - [primitives_autodetect_best]:  * optimized= 22
[10:10:08:159] [89418:00015d4b] [DEBUG][com.freerdp.primitives] - [primitives_autodetect_best]: primitives autodetect, using optimized
[10:10:08:259] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[10:10:08:259] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer <IP-address>
[10:10:08:306] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_tls_connect]: connecting to <rdgateway-domain>
Browse to: https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=a85cf173-4192-42f8-81fa-777a763e6e2c&response_type=code&scope=https%3A%2F%2Fwww.wvd.microsoft.com%2F.default&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient
Paste redirect URL here:
<redirect-url>
[10:10:12:228] [89418:00015d4b] [DEBUG][com.freerdp.utils.http] - [freerdp_http_request]: headers:
POST /common/oauth2/v2.0/token HTTP/1.1
Host: login.microsoftonline.com
Content-Type: application/x-www-form-urlencoded
Content-Length: 1039


[10:10:12:543] [89418:00015d4b] [DEBUG][com.freerdp.utils.http] - [freerdp_http_request]: body:
<body>
[10:10:13:307] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.http] - [http_chuncked_read]: chunked encoding end of stream received
[10:10:13:308] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Got HTTP Response data: {
  "Code": "E_PROXY_ORCHESTRATION_LB_SESSIONHOST_DEALLOCATED",
  "Message": "Session Host with ID = '<host>' has been deallocated we are starting the session host for you. Please retry connecting after 5 minutes.",
  "Target": null,
  "Details": null,
  "InnerError": null
}
[10:10:13:308] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Starting your VM. It may take up to 5 minutes
[10:10:13:308] [89418:00015d4b] [INFO][com.freerdp.client.common] - [client_common_retry_dialog]: [arm-transport] Starting your VM. It may take up to 5 minutes
[10:10:13:308] [89418:00015d4b] [WARN][com.freerdp.client.common] - [client_common_retry_dialog]: Automatic reconnection disabled, terminating. Try to connect again later
[10:10:13:311] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[10:10:13:311] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer <IP-address>
[10:10:13:391] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_tls_connect]: connecting to <rdgateway-domain>
[10:10:14:912] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.http] - [http_chuncked_read]: chunked encoding end of stream received
[10:10:14:912] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Got HTTP Response data: {
  "Code": "E_PROXY_ORCHESTRATION_LB_SESSIONHOST_DEALLOCATED",
  "Message": "Session Host with ID = '<host>' has been deallocated we are starting the session host for you. Please retry connecting after 5 minutes.",
  "Target": null,
  "Details": null,
  "InnerError": null
}
[10:10:14:913] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Starting your VM. It may take up to 5 minutes
[10:10:14:913] [89418:00015d4b] [WARN][com.freerdp.client.common] - [client_common_retry_dialog]: Automatic reconnection disabled, terminating. Try to connect again later
[10:10:14:916] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[10:10:14:916] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer <IP-address>
[10:10:14:989] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_tls_connect]: connecting to <rdgateway-domain>
[10:10:14:487] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.http] - [http_chuncked_read]: chunked encoding end of stream received
[10:10:14:488] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Got HTTP Response data: {
  "Code": "E_PROXY_ORCHESTRATION_LB_SESSIONHOST_DEALLOCATED",
  "Message": "Session Host with ID = '<host>' has been deallocated we are starting the session host for you. Please retry connecting after 5 minutes.",
  "Target": null,
  "Details": null,
  "InnerError": null
}
[10:10:14:488] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Starting your VM. It may take up to 5 minutes
[10:10:14:488] [89418:00015d4b] [WARN][com.freerdp.client.common] - [client_common_retry_dialog]: Automatic reconnection disabled, terminating. Try to connect again later
[10:10:14:491] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[10:10:14:491] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer <IP-address>
[10:10:14:565] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_tls_connect]: connecting to <rdgateway-domain>
[10:10:15:73] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.http] - [http_chuncked_read]: chunked encoding end of stream received
[10:10:15:74] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Got HTTP Response data: {
  "Code": "E_PROXY_ORCHESTRATION_LB_SESSIONHOST_DEALLOCATED",
  "Message": "Session Host with ID = '<host>' has been deallocated we are starting the session host for you. Please retry connecting after 5 minutes.",
  "Target": null,
  "Details": null,
  "InnerError": null
}
[10:10:15:74] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Starting your VM. It may take up to 5 minutes
[10:10:15:74] [89418:00015d4b] [WARN][com.freerdp.client.common] - [client_common_retry_dialog]: Automatic reconnection disabled, terminating. Try to connect again later
[10:10:15:77] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[10:10:15:77] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer <IP-address>
[10:10:15:140] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_tls_connect]: connecting to <rdgateway-domain>
[10:10:15:631] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.http] - [http_chuncked_read]: chunked encoding end of stream received
[10:10:15:631] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Got HTTP Response data: {
  "Code": "E_PROXY_ORCHESTRATION_LB_SESSIONHOST_DEALLOCATED",
  "Message": "Session Host with ID = '<host>' has been deallocated we are starting the session host for you. Please retry connecting after 5 minutes.",
  "Target": null,
  "Details": null,
  "InnerError": null
}
[10:10:15:631] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Starting your VM. It may take up to 5 minutes
[10:10:15:632] [89418:00015d4b] [WARN][com.freerdp.client.common] - [client_common_retry_dialog]: Automatic reconnection disabled, terminating. Try to connect again later
[10:10:15:635] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[10:10:15:635] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer <IP-address>
[10:10:15:684] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_tls_connect]: connecting to <rdgateway-domain>
[10:10:16:268] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.http] - [http_chuncked_read]: chunked encoding end of stream received
[10:10:16:269] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Got HTTP Response data: {
  "Code": "E_PROXY_ORCHESTRATION_LB_SESSIONHOST_DEALLOCATED",
  "Message": "Session Host with ID = '<host>' has been deallocated we are starting the session host for you. Please retry connecting after 5 minutes.",
  "Target": null,
  "Details": null,
  "InnerError": null
}
[10:10:16:269] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Starting your VM. It may take up to 5 minutes
[10:10:16:269] [89418:00015d4b] [WARN][com.freerdp.client.common] - [client_common_retry_dialog]: Automatic reconnection disabled, terminating. Try to connect again later
[10:10:16:272] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[10:10:16:272] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer <IP-address>
[10:10:16:349] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_tls_connect]: connecting to <rdgateway-domain>
[10:10:16:758] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.http] - [http_chuncked_read]: chunked encoding end of stream received
[10:10:16:758] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Got HTTP Response data: {
  "Code": "E_PROXY_ORCHESTRATION_LB_SESSIONHOST_DEALLOCATED",
  "Message": "Session Host with ID = '<host>' has been deallocated we are starting the session host for you. Please retry connecting after 5 minutes.",
  "Target": null,
  "Details": null,
  "InnerError": null
}
[10:10:16:758] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_bad_request]: Starting your VM. It may take up to 5 minutes
[10:10:16:758] [89418:00015d4b] [WARN][com.freerdp.client.common] - [client_common_retry_dialog]: Automatic reconnection disabled, terminating. Try to connect again later
[10:10:16:762] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[10:10:16:762] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer <IP-address>
[10:10:16:843] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_tls_connect]: connecting to <rdgateway-domain>
[10:10:18:694] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.http] - [http_chuncked_read]: chunked encoding end of stream received
[10:10:18:694] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_handle_request_ok]: Got HTTP Response data: {
  "udpPort": 0,
  "redirectedServerName": "<host>",
  "redirectedServerCert": "<server-cert>",
  "redirectedAuthBlob": "<auth-blob>",
  "redirectedAuthGuid": "<auth-guid>",
  "redirectedUsername": null,
  "clientUpdateLocation": "",
  "logonCertEnabled": false,
  "gatewayLocation": "<gateway-location>",
  "gatewayLocationPreWebSocket": "<gateway-location-pre-web-socket>",
  "gatewayConnectionToken": "<gateway-connection-token>",
  "gatewayLoadBalancerToken": "<gateway-load-balancer-token>",
  "gatewayLoadBalancerMethod": "Cookie",
  "clientRendezvousLocation": "<client-rendezvous-location>",
  "azureInstanceNetworkMetadata": "<azure-instance-network-metadata>",
  "iceServersConfig": {
    "turnEnabled": false,
    "stunEnabled": true,
    "stunServers": null,
    "turnServers": [
      {
        "url": {
          "secure": false,
          "host": "<host-ip>",
          "port": <port>,
          "transport": null
        },
        "username": "<username>",
        "password": "<password>",
        "realm": "\"rtcmedia\"",
        "expiresOn": "<expires-on>"
      }
    ]
  },
  "deviceId": "<device-id>",
  "deviceTenantId": "<device-tenant-id>",
  "sessionId": 0
}
[10:10:18:695] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.arm] - [arm_fill_gateway_parameters]: extracted target url <target-url>
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_set_negotiation_enabled]: Enabling security layer negotiation: TRUE
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_set_restricted_admin_mode_required]: Enabling restricted admin mode: FALSE
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_enable_rdp]: Enabling RDP security: FALSE
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_enable_tls]: Enabling TLS security: FALSE
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_enable_nla]: Enabling NLA security: FALSE
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_enable_ext]: Enabling NLA extended security: FALSE
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_enable_rdstls]: Enabling RDSTLS security: TRUE
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_enable_aad]: Enabling RDS AAD security: FALSE
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_INITIAL --> CONNECTION_STATE_NEGO
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_connect]: state: NEGO_STATE_RDSTLS
[10:10:18:702] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_attempt_rdstls]: Attempting RDSTLS security
[10:10:19:962] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[10:10:19:963] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer <IP-address>
[10:10:19:18] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.wst] - [wst_tls_connect]: connecting to <rdgateway-domain>
[10:10:19:188] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.wst] - [wst_handle_ok_or_forbidden]: Got Affinity cookie <cookie>
[10:10:19:192] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[10:10:19:192] [89418:00015d4b] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer <IP-address>
[10:10:19:237] [89418:00015d4b] [DEBUG][com.freerdp.core.gateway.wst] - [wst_tls_connect]: connecting to <rdgateway-domain>
[10:10:19:407] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_send_negotiation_request]: Adding terminating CRLF to routing token
[10:10:19:407] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_send_negotiation_request]: RequestedProtocols: 5
[10:10:19:555] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_process_negotiation_response]: RDP_NEG_RSP::flags = { [0x3f] |EXTENDED_CLIENT_DATA_SUPPORTED|DYNVC_GFX_PROTOCOL_SUPPORTED|RDP_NEGRSP_RESERVED|RESTRICTED_ADMIN_MODE_SUPPORTED|REDIRECTED_AUTHENTICATION_MODE_SUPPORTED|UNKNOWN }
[10:10:19:555] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_recv]: selected_protocol: 4
[10:10:19:555] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_attempt_rdstls]: state: NEGO_STATE_FINAL
[10:10:19:555] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_connect]: Negotiated RDSTLS security
[10:10:19:555] [89418:00015d4b] [DEBUG][com.freerdp.core.nego] - [nego_try_connect]: nego_security_connect with PROTOCOL_RDSTLS
[10:10:19:827] [89418:00015d4b] [WARN][com.freerdp.crypto] - [verify_cb]: Certificate verification failure 'unable to get local issuer certificate (20)' at stack position 0
[10:10:19:827] [89418:00015d4b] [WARN][com.freerdp.crypto] - [verify_cb]: CN = <cn>
[10:10:19:831] [89418:00015d4b] [DEBUG][com.freerdp.core.rdstls] - [rdstls_set_state]: -- RDSTLS_STATE_INITIAL  --> RDSTLS_STATE_CAPABILITIES
[10:10:19:831] [89418:00015d4b] [DEBUG][com.freerdp.core.rdstls] - [rdstls_set_state]: -- RDSTLS_STATE_CAPABILITIES     --> RDSTLS_STATE_AUTH_REQ
[10:10:19:832] [89418:00015d4b] [DEBUG][com.freerdp.core.rdstls] - [rdstls_set_state]: -- RDSTLS_STATE_AUTH_REQ --> RDSTLS_STATE_AUTH_RSP
[10:10:20:983] [89418:00015d4b] [DEBUG][com.freerdp.core.rdstls] - [rdstls_set_state]: -- RDSTLS_STATE_AUTH_RSP --> RDSTLS_STATE_FINAL
[10:10:20:984] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_NEGO --> CONNECTION_STATE_MCS_CREATE_REQUEST
[10:10:20:984] [89418:00015d4b] [DEBUG][com.freerdp.core.gcc] - [gcc_write_client_core_data]: Sending highColorDepth=HIGH_COLOR_24BPP, supportedColorDepths=RNS_UD_32BPP_SUPPORT|RNS_UD_24BPP_SUPPORT|RNS_UD_16BPP_SUPPORT|RNS_UD_15BPP_SUPPORT, earlyCapabilityFlags=RNS_UD_CS_WANT_32BPP_SESSION|RNS_UD_CS_VALID_CONNECTION_TYPE|RNS_UD_CS_SUPPORT_NETCHAR_AUTODETECT|RNS_UD_CS_SUPPORT_DYNVC_GFX_PROTOCOL|RNS_UD_CS_SUPPORT_HEARTBEAT_PDU|RNS_UD_CS_SUPPORT_SKIP_CHANNELJOIN|[0x00000da2]
[10:10:20:984] [89418:00015d4b] [DEBUG][com.freerdp.core.gcc] - [gcc_write_client_monitor_data]: MonitorCount=1
[10:10:20:984] [89418:00015d4b] [DEBUG][com.freerdp.core.gcc] - [gcc_write_client_monitor_data]: FINISHED
[10:10:20:984] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_MCS_CREATE_REQUEST --> CONNECTION_STATE_MCS_CREATE_RESPONSE
[10:10:20:136] [89418:00015d4b] [DEBUG][com.freerdp.core.gcc] - [gcc_read_server_core_data]: Received EarlyCapabilityFlags=RNS_UD_CS_SUPPORT_STATUSINFO_PDU|RNS_UD_CS_STRONG_ASYMMETRIC_KEYS|[0x0000000c]
[10:10:20:136] [89418:00015d4b] [DEBUG][com.freerdp.core.gcc] - [gcc_read_server_security_data]: Server rdp encryption method: NONE
[10:10:20:136] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_MCS_CREATE_RESPONSE --> CONNECTION_STATE_MCS_ERECT_DOMAIN
[10:10:20:137] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_MCS_ERECT_DOMAIN --> CONNECTION_STATE_MCS_ATTACH_USER
[10:10:20:137] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_MCS_ATTACH_USER --> CONNECTION_STATE_MCS_ATTACH_USER_CONFIRM
[10:10:20:278] [89418:00015d4b] [DEBUG][com.freerdp.core.connection] - [rdp_client_skip_mcs_channel_join]:  rdpdr [1004]
[10:10:20:278] [89418:00015d4b] [DEBUG][com.freerdp.core.connection] - [rdp_client_skip_mcs_channel_join]:  rdpsnd [1005]
[10:10:20:278] [89418:00015d4b] [DEBUG][com.freerdp.core.connection] - [rdp_client_skip_mcs_channel_join]:  drdynvc [1006]
[10:10:20:278] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_MCS_ATTACH_USER_CONFIRM --> CONNECTION_STATE_SECURE_SETTINGS_EXCHANGE
[10:10:20:278] [89418:00015d4b] [DEBUG][com.freerdp.core.info] - [rdp_write_info_packet]: Client Info Packet Flags = INFO_MOUSE|INFO_DISABLECTRLALTDEL|INFO_AUTOLOGON|INFO_UNICODE|INFO_MAXIMIZESHELL|INFO_LOGONNOTIFY|INFO_COMPRESSION|INFO_ENABLEWINDOWSKEY|INFO_FORCE_ENCRYPTED_CS_PDU|INFO_LOGONERRORS|INFO_MOUSE_HAS_WHEEL|INFO_AUDIOCAPTURE
[10:10:20:283] [89418:00015d4b] [DEBUG][com.winpr.timezone] - [winpr_get_unix_timezone_identifier_from_file]: tzid: Europe/Kiev
[10:10:20:283] [89418:00015d4b] [DEBUG][com.winpr.timezone] - [winpr_get_timezone_from_link]: tzid: Europe/Kyiv
[10:10:20:283] [89418:00015d4b] [INFO][com.winpr.timezone] - [winpr_detect_windows_time_zone]: tzid: Europe/Kyiv
[10:10:20:286] [89418:00015d4b] [ERROR][com.winpr.timezone] - [winpr_detect_windows_time_zone]: Unable to find a match for unix timezone: Europe/Kyiv
[10:10:20:286] [89418:00015d4b] [DEBUG][com.winpr.timezone] - [GetTimeZoneInformation]: tz not found, using computed bias -180.
[10:10:20:286] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [rdp_write_system_time]: Time: y=0,m=0,dow=0,d=0, 00:00:00.000
[10:10:20:286] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [rdp_write_client_time_zone]: bias=-180
[10:10:20:286] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [rdp_write_client_time_zone]: StandardName:
[10:10:20:286] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [winpr_HexLogDump]: 0000 43 00 00 00 6c 00 00 00 69 00 00 00 65 00 00 00 C...l...i...e...
[10:10:20:286] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [winpr_HexLogDump]: 0016 6e 00 00 00 74 00 00 00 20 00 00 00 4c 00 00 00 n...t... ...L...
[10:10:20:286] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [winpr_HexLogDump]: 0032 6f 00 00 00 63 00 00 00 61 00 00 00 6c 00 00 00 o...c...a...l...
[10:10:20:286] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [winpr_HexLogDump]: 0048 20 00 00 00 54 00 00 00 69 00 00 00 6d 00 00 00  ...T...i...m...
[10:10:20:286] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [winpr_HexLogDump]: [length=64]
[10:10:20:286] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [rdp_write_client_time_zone]: DaylightName:
[10:10:20:287] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [winpr_HexLogDump]: 0000 43 00 00 00 6c 00 00 00 69 00 00 00 65 00 00 00 C...l...i...e...
[10:10:20:287] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [winpr_HexLogDump]: 0016 6e 00 00 00 74 00 00 00 20 00 00 00 4c 00 00 00 n...t... ...L...
[10:10:20:287] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [winpr_HexLogDump]: 0032 6f 00 00 00 63 00 00 00 61 00 00 00 6c 00 00 00 o...c...a...l...
[10:10:20:287] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [winpr_HexLogDump]: 0048 20 00 00 00 54 00 00 00 69 00 00 00 6d 00 00 00  ...T...i...m...
[10:10:20:287] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [winpr_HexLogDump]: [length=64]
[10:10:20:287] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [rdp_write_client_time_zone]: StandardBias=0
[10:10:20:287] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [rdp_write_system_time]: Time: y=0,m=0,dow=0,d=0, 00:00:00.000
[10:10:20:287] [89418:00015d4b] [DEBUG][com.freerdp.core.timezone] - [rdp_write_client_time_zone]: DaylightBias=0
[10:10:20:287] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_SECURE_SETTINGS_EXCHANGE --> CONNECTION_STATE_CONNECT_TIME_AUTO_DETECT_REQUEST
[10:10:20:423] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_CONNECT_TIME_AUTO_DETECT_REQUEST --> CONNECTION_STATE_LICENSING
[10:10:20:423] [89418:00015d4b] [INFO][com.freerdp.core.license] - [license_client_recv]: Receiving ERROR_ALERT Packet
[10:10:20:423] [89418:00015d4b] [WARN][com.freerdp.core.license] - [license_read_binary_blob_data]: license binary blob::type BB_ERROR_BLOB, length=0, skipping.
[10:10:20:424] [89418:00015d4b] [DEBUG][com.freerdp.core.license] - [license_read_error_alert_packet]: dwErrorCode: STATUS_VALID_CLIENT, dwStateTransition: ST_NO_TRANSITION
[10:10:20:424] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_LICENSING --> CONNECTION_STATE_MULTITRANSPORT_BOOTSTRAPPING_REQUEST
[10:10:20:427] [89418:00015d4b] [WARN][com.freerdp.core.multitransport] - [multitransport_recv_request]: reserved is 1 instead of 0, skipping 2125bytes of unknown data
[10:10:21:234] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_MULTITRANSPORT_BOOTSTRAPPING_REQUEST --> CONNECTION_STATE_CAPABILITIES_EXCHANGE_DEMAND_ACTIVE
[10:10:21:234] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_read_share_control_header][0x61f000000080]: type=PDU_TYPE_DEMAND_ACTIVE[0x00000001], tpktLength=457, remainingLength=451
[10:10:21:239] [89418:00015d4b] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x61f000000080]: CONNECTION_STATE_CAPABILITIES_EXCHANGE_DEMAND_ACTIVE --> CONNECTION_STATE_CAPABILITIES_EXCHANGE_CONFIRM_ACTIVE
```

</details>

**Comment by akallabeth** [comment]
@progxaker ok, my mistake. Fixed with #9403 
we should actually abort if the callback tells us so :grin:

**Comment by progxaker** [comment]
Hello. After fixes #9400 and #9416, FreeRDP successfully connects to an AVD instance w/ and w/o `/auto-reconnect` as expected.

Thank you @akallabeth :tada: 

Without `/auto-reconnect` log:
```
[08:24:45:132] [93158:00016be7] [INFO][com.freerdp.client.common] - [client_common_retry_dialog]: [arm-transport] Starting your VM. It may take up to 5 minutes                                   
[08:24:45:132] [93158:00016be7] [WARN][com.freerdp.client.common] - [client_common_retry_dialog]: Automatic reconnection disabled, terminating. Try to connect again later                        
[08:24:45:134] [93158:00016be7] [ERROR][com.freerdp.core.nego] - [nego_connect]: Protocol Security Negotiation Failure                                                                            
[08:24:45:134] [93158:00016be7] [ERROR][com.freerdp.core] - [rdp_client_connect]: ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x0002000C]                                                            
[08:24:45:134] [93158:00016be7] [ERROR][com.freerdp.core.connection] - [rdp_client_connect]: Error: protocol security negotiation or connection failure
```
